### PR TITLE
Added all the ingredients to enforce high quality code

### PR DESCRIPTION
**Pull request by AndPotap**

### CODE REVIEW

**Comment by mfinzi** [comment]
Can you explain what it does?

**Comment by AndPotap** [comment]
@mfinzi , I'm in the process of fixing all the code. I'll explain once I'm done

**Comment by AndPotap** [comment]
@mfinzi Basically. (1) I added a bash file to add a pre-commit hook for each user so that the formatters are run when they make a commit. (2) Created a Github action that checks both that all the files follow the format and that it follows PEP8 (via flake8). (3) Run the formater on all the files in the repo. (4) Fixed all the errors highlighted by flake8: unsued imports, multiple imports, unsued variables, trailing white spaces, undefined variables, etc.

**Comment by AndPotap** [comment]
@mfinzi could you accept it to give it a try :upside_down_face:

File: cola/utils_test.py [hunk 1]
```diff
@@ -79,41 +79,49 @@ def __init__(self, *cases, ids=None):
...
     def excluding(self):
-        self.indexing=False
+        self.indexing = False
         self.indexed_cases = set()
         return self
 
     def __getitem__(self, indexed_cases):
-        if isinstance(indexed_cases,tuple) and len(indexed_cases)>1: # multiple arguments, need to use cross product
-            expanded_indexed_cases = [index(c,t) for t,c in zip(indexed_cases,self.cases)]
+        if isinstance(
```
Comment (mfinzi) [inline cola/utils_test.py]: this is not better

Comment (mfinzi) [reply cola/utils_test.py]: I guess we should just get in the habit of putting the comments above?

File: setup.cfg [hunk 2]
```diff
@@ -8,6 +8,10 @@ BLANK_LINE_BEFORE_NESTED_CLASS_OR_DEF = False
 max-line-length = 100
 ignore = E301
 
+[flake8]
+max-line-length = 100
```
Comment (mfinzi) [inline setup.cfg]: how about 120?

**Review by mfinzi — APPROVED** [review]

File: cola/utils_test.py (hunk 1 above)
Comment (AndPotap) [reply cola/utils_test.py]: Yeah good catch. Let me fix this. It is because of the comment indeed. I'll put it above

File: setup.cfg (hunk 2 above)
Comment (AndPotap) [reply setup.cfg]: I prefer 120 but I'm not sure if this is too far from what people are used to (80)

Comment (mfinzi) [reply setup.cfg]: let's do 120. We are the 👑

**Pull request merged**